### DISCUSSION TITLE
[Possible Feature Request] Color balance in imagemagick as featured in Krita

### DISCUSSION THREAD

**Discussion by developer1105** [ImageMagick/ImageMagick#6986] [Development]
### ImageMagick version

7.1.1-24

### Operating system, version and so on

Arch Linux x86_64 / kernel 6.1.69-1-lts

### Description

How to do in imagemagick an effect similar to color balance in Krita?

Specific details: 
1) Open Krita
2) Menu -> Filter|Adjust|'Color Balance'
3) Dialog box will open
4) item Shadows, set the scale Yellow-Blue to 100
5) item Midtones, set the scale Yellow-Blue to 100 
6) item Highligts, set the scale Yellow-Blue to 100
7) option 'Preserve Luminosity' checked (on)

If this is not achievable in the current version, I would like to suggest it to be implemented in a future version, if the developers find this interesting and if it is kind of simple to do.

**Comment by snibgo** [comment]
It can probably be done in IM as a sequence of operations. One operation might do the equivalent of "set the scale Yellow-Blue to 100". You could have an operation that makes masks that correspond to shadows, midtones and highlights. However, as you apply the same operation to all three areas, that probably isn't needed.

Without the technical details of what "set the scale Yellow-Blue to 100" does, advice is difficult.

**Comment by developer1105** [comment]
I was adding noise (dithering) to reduce color banding in an image, and then by chance, started tinkering with 'Color Balance'.

Then I noticed that text in black-and-white images (shades of gray to be more precise) were more easily readble and caused less eye strain with this procedure. 

I would like to do it with hundreds of images, since it would be overkill in Krita.
